Client/src/Components/DatasetUpload/DataSection/VariableHeader.tsx

```diff
@@ -1,19 +1,20 @@
 import { Box, Typography } from '@material-ui/core'
+import React, { useContext } from 'react'
 import { getDimensionNameById, getUnitNameById } from '../../../Common/Helpers/DimensionHelpers'
 
-import { IDimensionModel } from '../../../Models/Dimensions/IDimensionModel'
 import { IVariable } from '../../../Models/Datasets/IDatasetModel'
-import React from 'react'
+import { StoreContext } from '../../../Context/StoreContext'
 
 interface IProps {
   variable: IVariable,
   index: number,
   onHeaderClick: (index: number) => void
-  dimensions: IDimensionModel[]
 }
 
 export const VariableHeader = (props: IProps) => {
-  const { variable, index, onHeaderClick, dimensions } = { ...props }
+  const { variable, index, onHeaderClick } = { ...props }
+  const { dimensions } = useContext(StoreContext).store.getPreloadedData()
```

Comment:
Much cleaner

Reply:
yeah we can just store the user in the store, didnt do it cause felt like too many changes already, but I'll do it

Reply:
will implement this in the pr right after cause it is a lot of new changes that will get mixed up with the changes of this PR